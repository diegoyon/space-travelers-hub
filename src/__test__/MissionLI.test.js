import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../redux/configureStore';
import MissionLi from '../components/missions';

test('check snapshot of missions list', () => {
  const component = render(
    <React.StrictMode>
      <Provider store={store}>
        <Router>
          <MissionLi
            name="mission"
            description="new description"
            reserved={false}
            id="1"
          />
        </Router>
      </Provider>
    </React.StrictMode>,
  );
  expect(component).toMatchSnapshot();
});

test('check snapshot of missions list', () => {
  const component = render(
    <React.StrictMode>
      <Provider store={store}>
        <Router>
          <MissionLi
            name="mission"
            description="new description"
            reserved={false}
            id="2"
          />
        </Router>
      </Provider>
    </React.StrictMode>,
  );
  expect(component).toMatchSnapshot();
});
