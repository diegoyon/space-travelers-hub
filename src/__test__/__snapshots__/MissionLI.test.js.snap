// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check snapshot of missions list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="mission-li"
      >
        <p
          class="mission-name"
        >
          mission
        </p>
        <p
          class="mission-description"
        >
          new description
        </p>
        <div
          class="mission-status"
        >
          <p
            class=""
          >
            not a member
          </p>
        </div>
        <div
          class="btn-container"
        >
          <button
            type="button"
          >
            join
          </button>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="mission-li"
    >
      <p
        class="mission-name"
      >
        mission
      </p>
      <p
        class="mission-description"
      >
        new description
      </p>
      <div
        class="mission-status"
      >
        <p
          class=""
        >
          not a member
        </p>
      </div>
      <div
        class="btn-container"
      >
        <button
          type="button"
        >
          join
        </button>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`check snapshot of missions list 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="mission-li"
      >
        <p
          class="mission-name"
        >
          mission
        </p>
        <p
          class="mission-description"
        >
          new description
        </p>
        <div
          class="mission-status"
        >
          <p
            class=""
          >
            not a member
          </p>
        </div>
        <div
          class="btn-container"
        >
          <button
            type="button"
          >
            join
          </button>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="mission-li"
    >
      <p
        class="mission-name"
      >
        mission
      </p>
      <p
        class="mission-description"
      >
        new description
      </p>
      <div
        class="mission-status"
      >
        <p
          class=""
        >
          not a member
        </p>
      </div>
      <div
        class="btn-container"
      >
        <button
          type="button"
        >
          join
        </button>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`check snapshot of missions list 2 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="mission-li"
      >
        <p
          class="mission-name"
        >
          mission 2
        </p>
        <p
          class="mission-description"
        >
          new description 2
        </p>
        <div
          class="mission-status"
        >
          <p
            class="joined"
          >
            active member
          </p>
        </div>
        <div
          class="btn-container joined"
        >
          <button
            type="button"
          >
            leave
          </button>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="mission-li"
    >
      <p
        class="mission-name"
      >
        mission 2
      </p>
      <p
        class="mission-description"
      >
        new description 2
      </p>
      <div
        class="mission-status"
      >
        <p
          class="joined"
        >
          active member
        </p>
      </div>
      <div
        class="btn-container joined"
      >
        <button
          type="button"
        >
          leave
        </button>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
